lib/ddtrace/contrib/sinatra/framework.rb

```diff
@@ -0,0 +1,78 @@
+module Datadog
+  module Contrib
+    # Instrument Sinatra.
+    module Sinatra
+      # Sinatra framework code, used to essentially:
+      # - handle configuration entries which are specific to Datadog tracing
+      # - instrument parts of the framework when needed
+      module Framework
+        # Configure Rack from Sinatra, but only if Rack has not been configured manually beforehand
+        def self.setup
+          Datadog.configure do |datadog_config|
+            sinatra_config = config_with_defaults(datadog_config)
+            activate_rack!(datadog_config, sinatra_config) unless Datadog.configuration.instrumented_integrations.key?(:rack)
+          end
```

Comment:
Because `Datadog.configure` recreates all tracing internals all over again, it's good to avoid it if possible.

In this case, the conditional execution can be moved to outside the block to prevent tracer reinitialization here:

```suggestion
            activate_rack!(datadog_config, sinatra_config)
          end unless Datadog.configuration.instrumented_integrations.key?(:rack)
```